test/definition/component-info.test.ts

```diff
@@ -4,65 +4,148 @@ import { generateComponentSetInfo, processInfo } from '../../lib/definition/comp
 import { processTemplates } from '../../lib/definition/process-templates';
 import { ComponentsDefinition } from '../../lib/models';
 
-describe('Component Info', () => {
-    const componentPath = path.resolve('./test/resources/minimal-sample');
-    let definition: ComponentsDefinition;
-    beforeEach(async () => {
-        definition = JSON.parse(
-            (await fs.promises.readFile(path.join(componentPath, 'components-definition.json'))).toString(),
-        );
-    });
+describe('processInfo', () => {
+    describe('minimal component set', () => {
+        const componentPath = path.resolve('./test/resources/minimal-sample');
+        let definition: ComponentsDefinition;
+        beforeEach(async () => {
+            definition = JSON.parse(
+                (await fs.promises.readFile(path.join(componentPath, 'components-definition.json'))).toString(),
+            );
+        });
 
-    const renditionResolver = async (relativePath: string) => {
-        const file = path.join(componentPath, relativePath);
-        try {
-            return (await fs.promises.readFile(file)).toString();
-        } catch (e) {
-            return;
+        const renditionResolver = async (relativePath: string) => {
+            const file = path.join(componentPath, relativePath);
+            try {
+                return (await fs.promises.readFile(file)).toString();
+            } catch (e) {
+                return;
+            }
+        };
+
+        async function processTemplatesFromZip() {
+            await processTemplates(renditionResolver, definition);
         }
-    };
 
-    async function processTemplatesFromZip() {
-        await processTemplates(renditionResolver, definition);
-    }
+        async function processEmptyTemplates() {
+            await processTemplates(async (_relativePath: string) => {
+                return '<div></div>';
+            }, definition);
+        }
+
+        describe('generateComponentSetInfo', () => {
+            it('should return the component set information data', async () => {
+                expect(await generateComponentSetInfo(definition, renditionResolver)).toEqual({
+                    components: {
+                        body: {
+                            fields: [
+                                {
+                                    contentKey: 'text',
+                                    type: 'editable',
+                                },
+                            ],
+                        },
+                        intro: {
+                            fields: [
+                                {
+                                    contentKey: 'text',
+                                    type: 'editable',
+                                },
+                            ],
+                        },
+                    },
+                });
+            });
+        });
+
+        describe('processInfo', () => {
```

Comment:
Another processInfo describe block? That does not make sense. You should remove one.

Reply:
Good point. Missed that one during a merge. Fixed it.